google/ggrc-core
# GGRC-8311 Move hide/restore logic out of modals-controller

# Issue description

`modals-controller` contains the logic related to hiding/restoring fields. It can be moved to the separate component which should be used where this logic is needed.

# Steps to test the changes

1. Open any modal with hide/restore functionality.

Expected result: Hide/restore logic works correctly, after saving the state of hidden fields are saved in local storage like it was before the change.

# Solution description

Created `<modal-container>` component, which contains the logic related to hide/restore.

Additional changes:
Used static import for one template in modals-controller.
GCAs initialisation is moved to <modal-container> (since GCA can be hidden).

NOTES - the following existing issues are NOT fixed in the scope of the current PR:

1) Currently, states of GCAs (hidden) is not saved properly (on `dev` env).

2) For assessment_templates/modal-content, several hidable fields state is not saved in local storage. When hide via Hide all button, then it’s not working properly as well.

3) On Program, Regulations, Requirements, Tasks, Threats modals - Custom roles don’t have hide button, but they can be hidden via Hide all button

4) Task and Task Group doesn’t hide Code (Slug) field after saving if Hide button was clicked.

# Sanity checklist




# PR Review checklist
